### ISSUE TITLE
Investigate options to make partial backfills resumable more often

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#1944]
As far as I understand, we currently update the version of backfilled data only after a whole shard has completed backfilling.
If for some reason a node goes down while backfilling, it has to backfill the whole shard again when it gets restarted. In combination with slow backfills on rotational drives (#1885), that can become quite a pain.

I wonder if we could artificially split up a range shard into a number of (even) smaller ranges during backfilling. Each of those subranges could be backfilled independently one after another. As soon as one subrange finishes backfilling, its current data version could be updated.
If a partially complete backfilling process is aborted for some reason, subranges that had already completed backfilling wouldn't have to be backfilled all over again when the process is resumed later.

Marking for subsequent. If this turns out to be difficult, we should probably move this to 2.x. If it is easy to do on the other hand it might be worth doing earlier.
There might also be other options.

**Comment by danielmewes** [comment]
Superseded by https://github.com/rethinkdb/rethinkdb/issues/3886
